### PR TITLE
[WASM] Add full support of RenderTransform in TransformToVisual

### PR DESCRIPTION
**Pull request by dr1rrb**
GitHub Issue (If applicable): https://github.com/unoplatform/uno/issues/2035

## BugFix & Feature
Add full support of the `RenderTransform` in the `UIElement.TransformToVisual`

## What is the current behavior?
The `UIElement.TransformToVisual` only request the current bounds of the UIElements and makes a diff to get an approximation of the offsetX and offsetY, but does not take in consideration the possible scale in between, nor the rotation.

On WASM this impacts the pointers coordinates as they are [computed from absolute coordinates](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.wasm.cs#L53) and should be "scaled down" in order to remain in the bounds of the control.

## What is the new behavior?
The returned transform is now including scale and rotation applied by any parent control.

## PR Checklist
- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

**Comment by dr1rrb** [comment]
> 
> 
> Needs tests

Obviously! I will also add some documentation :) But before I'll see if I can multi target this method

**Comment by carldebilly** [comment]
@davidjohnoliver @jeromelaban @ghuntley 
I think we should'nt comment about missing tests, commented code or missing documentation when it's still a draft PR.

**Comment by davidjohnoliver** [comment]
> @davidjohnoliver @jeromelaban @ghuntley
> I think we should'nt comment about missing tests, commented code or missing documentation when it's still a draft PR.

I'm not sure I see the point of draft PRs.

**Comment by carldebilly** [comment]
A draft PR is a way to make sure it's working with CI and get resulting nuget packages for more testing. It's also a way to start a discussion about eventual changes.

**Comment by dr1rrb** [comment]
> 
> 
> > @davidjohnoliver @jeromelaban @ghuntley
> > I think we should'nt comment about missing tests, commented code or missing documentation when it's still a draft PR.
> 
> I'm not sure I see the point of draft PRs.

In that case it was to get your review on the fonctionnal aspect of this PR, especially as we chat about this specific method in one of your recent PRs :)

**Comment by paulovila** [comment]
Thanks @dr1rrb , still no joy, I checked out dev/dr/TouchTrans
I overwrote C:\Repos\Uno\src\crosstargeting_override.props
with   <UnoNugetOverrideVersion>2.0.512-dev.9999</UnoNugetOverrideVersion>

generated
  C:\Users\pvila\.nuget\packages\Uno.UI\2.0.512-dev.9999\lib\netstandard2.0 and C:\Users\pvila\.nuget\packages\Uno.UI\2.0.512-dev.9999\ref\netstandard2.0

Still same behaviour buttons do not generate the command 
 https://github.com/SuperJMN/UnoSimpleCalc

**Comment by dr1rrb** [comment]
Note: this PR only targets WASM, and a issue was opened to multi-target this for other platforms  (https://github.com/unoplatform/uno/issues/2114)

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml/UIElementTests/UIElementTests.cs [hunk 1]
```diff
@@ -5,71 +5,34 @@
...
 using NUnit.Framework;
 using SamplesApp.UITests.TestFramework;
+using Uno.UITest.Helpers.Queries;
 
 namespace SamplesApp.UITests.Windows_UI_Xaml
 {
 	[TestFixture]
 	public class UIElementTests : SampleControlUITestBase
 	{
+		// TODO: convert this to RuntimeTests
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml/UIElementTests/UIElementTests.cs]: issue # ?

File: doc/articles/features/routed-events.md [hunk 2]
```diff
@@ -159,19 +159,17 @@ These _routed events_ are not implemented yet in Uno:
 * `DragOver`
 * `Drop`
 * `Holding`
-* `ManipulationCompleted`
-* `ManipulationDelta`
 * `ManipulationInertiaStarting`
-* `ManipulationStarted`
-* `ManipulationStarting`
```
Comment (carldebilly) [inline doc/articles/features/routed-events.md]: You remvoed them from the "unsupported" section, but they should appear now in the "supported" section :-)

File: src/Uno.UI/UI/Xaml/UIElement.wasm.cs [hunk 3]
```diff
@@ -601,31 +604,70 @@ public override string ToString()
...
+				if (transform == null)
+				{
+					// As this is the common case, avoid Matrix computation when a basic addition is sufficient
+					offsetX += elt._nativeLayoutSlot.X;
+					offsetY += elt._nativeLayoutSlot.Y;
+				}
+				else
+				{
+					// First apply any pending arrange offset that would have been impacted by this RenderTransform (eg. scaled)
+					// Friendly reminder: Matrix multiplication is usually not commutative ;)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElement.wasm.cs]: 🤣

File: src/Uno.UI/UI/Xaml/UIElement.wasm.cs [hunk 4]
```diff
@@ -601,31 +604,70 @@ public override string ToString()
...
-				Matrix = new Matrix(
-					m11: 1,
-					m12: 0,
-					m21: 0,
-					m22: 1,
-					offsetX: bounds.X - otherBounds.X,
-					offsetY: bounds.Y - otherBounds.Y
-				)
-			};
+				// Unfortunately we didn't found the 'visual' in our parent hierarchy,
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElement.wasm.cs]:
```suggestion
				// Unfortunately we didn't find the 'visual' in our parent hierarchy,
```

File: src/Uno.UI/Uno.UI.csproj [hunk 5]
```diff
@@ -347,6 +347,14 @@
 		</ProjectReference>
 	</ItemGroup>
 
+	<ItemGroup>
```
Comment (davidjohnoliver) [inline src/Uno.UI/Uno.UI.csproj]: don't merge this

File: doc/articles/features/routed-events.md (hunk 2 above)
Comment (dr1rrb) [reply doc/articles/features/routed-events.md]: Indeed, I marked them as implemented at the bottom of the file when I did it, but I forgot to update the table at the top of the file ... done!

**Pull request merged**